fission-web-server/library/Fission/Web/Server/IPFS/Pinner/Class.hs

```diff
@@ -0,0 +1,17 @@
+module Fission.Web.Server.IPFS.Pinner.Class (MonadIPFSPinner (..)) where
+
+import           Fission.Prelude
+
+import qualified Fission.Web.Server.IPFS.Cluster.Error as Cluster
+
+import qualified Network.IPFS.Add.Error                as IPFS.Pin
+import qualified Network.IPFS.Types                    as IPFS
+
+type Errors' = OpenUnion
+  '[ IPFS.Pin.Error
+   , Cluster.Error
+   ]
```

Comment:
Protip, but _no need to change here_.

With `OpenUnion`, we can have a bunch of separate errors that get composed as needed. So, for example:

```haskell
newtype ClusterDaemonErr = ClusterDaemonErr Text
newtype UnknownPinErr = UnknownPinErr    Text
newtype UnexpectedOutput = UnexpectedOutput Text

type ClusterError = '[ClusterDaemonErr, UnknownPinErr, UnexpectedOutput]
type IPFSError = '[...]

type Errors' = OpenUnion (ClusterError ++ IPFSErrors)
```

Now we can reuse `UnexpectedOutput` in a bunch of contexts, without needing to make `UnexpectedIPFSOutput`, `UnexpectedClusterOutput`, `UnexpectedServerOutput`, and so on

Reply:
Oh cool :ok_hand: :ok_hand: